src/components/staffing/HeaderCell.js

```diff
@@ -10,6 +10,10 @@ export default class HeaderCell extends React.Component {
   };
 
   render() {
+    const addEllipsisToLongName = (name) => {
+      return name.length > 21 ? `${name.slice(0, 21)}...` : name
```

Comment:
Add ellipsis if longer than 21 characters, which theoretically should work as the table cells are of fixed width and font-size is set

Issues with this approach is that if we ever change column width or font size in future must also change this value - attempted to limit the width of the tag containing name but the children of table cell do not respect its width oddly so could not get it to work.

Please give feedback on this method / any possible better ways if you can!

Reply:
High level: the function looks good! Would have done the same!

But if we zoom in there are a few things to improve:
- the function is declared in the "render" block → this means it gets re-created every time there is a render, that's not very efficient and although it does not have a huge impact, it's a bad practise that could have real impact if it spread to everywhere in the codebase
- you could make it a re-usable util that accepts a `value: string` and a `maxSize: number` that defaults to 21 to truncate the value (and this could be re-used for project names)

Reply:
Sounds good, thanks John I will make the revision now 👍

Reply:
Revised in latest commit